### ISSUE TITLE
Flutter 1.22 Failed assertion: boolean expression must not be null

### ISSUE THREAD

**Issue by amorenew** [rrousselGit/freezed#276]
**Describe the bug**
After Flutter upgrading to 1.22 I got the following error
Failed assertion: boolean expression must not be null

**To Reproduce**

My pub spec
```
environment:
  sdk: ">=2.10.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  # UI thingies
  cupertino_icons: ^1.0.0
  flare_splash_screen: ^3.0.1
  pin_code_fields: ^5.1.0
  flushbar: ^1.10.4
  flutter_spinkit: ^4.1.2+1
  flutter_svg: ^0.17.1
  flutter_advanced_networkimage: ^0.7.0
  # BLoC for state management
  flutter_bloc: ^6.0.0
  # Service locator
  get_it: ^4.0.4
  injectable: ^1.0.4
  # Functional programming thingies
  dartz: ^0.9.1
  # Annotations for the freezed code-generator
  freezed_annotation: ^0.12.0
  # Util
  get_version: ^0.2.2
  native_updater: ^0.0.3
  archive: ^2.0.13
  intl: ^0.16.1
  connectivity: ^0.4.9+3
  trust_fall: ^1.0.4
  random_string: ^2.1.0
  url_launcher: ^5.7.2
  flutter_local_notifications: ^1.4.4+2
  retry: ^3.0.1
  json_serializable: ^3.5.0
  logging: ^0.11.4
  # Local Database
  hive: ^1.4.1+1
  path_provider: ^1.6.18
  path: ^1.7.0
  # Remote API
  dio: ^3.0.10
  # Firebase related
  firebase_core: ^0.5.0
  firebase_messaging: ^7.0.0
  firebase_performance: ^0.4.0
  firebase_crashlytics: ^0.2.0
  # Location related
  google_maps_flutter: ^0.5.33
  flutter_polyline_points: ^0.2.2
  background_locator: ^1.2.2+1
  location_permissions: ^3.0.0+1
  geohash: ^0.2.1
  flutter_geofence: ^0.2.0
  location: ^3.0.2
  wakelock: ^0.1.4+2
  multi_image_picker: ^4.7.14
  progress_indicators: ^0.1.5
  keyboard_actions: "^3.3.0"
  device_info: ^0.4.2+7
  device_preview: ^0.4.8
  analyzer: ^0.39.2

dependency_overrides:
  flutter_svg: ^0.19.0
  freezed_annotation: ^0.12.0
  analyzer: ^0.40.1
  #platform: 3.0.0-nullsafety.2
  #process: 4.0.0-nullsafety.2
  #file: 6.0.0-nullsafety.2

dev_dependencies:
  # Test related
  flutter_test:
    sdk: flutter
  mockito: ^4.1.1
  flutter_driver:
    sdk: flutter
  test: any
  #screenshots: ^2.1.1
  # Code-generator related
  build_runner: ^1.10.3
  freezed: ^0.12.1
  injectable_generator: ^1.0.4
  # Custom linting rules
  lint: ^1.3.0
  assets_scanner: ^1.1.1
```

My class:
```
import 'package:freezed_annotation/freezed_annotation.dart';

part 'failures.freezed.dart';

@freezed
abstract class ValueFailure<T> with _$ValueFailure<T> {
  const factory ValueFailure.invalidPin({
    @nullable @required String failedValue,
  }) = InvalidPin<T>;
  const factory ValueFailure.shortPin({
    @nullable @required String failedValue,
  }) = ShortPin<T>;
}
```
**Expected behavior**
Build should not fail
`flutter pub run build_runner build --delete-conflicting-outputs`

**Comment by salehahmedZ** [comment]
try to remove `analyzer: ^0.40.1` from `dependency_overrides:` section

**Comment by amorenew** [comment]
@salehahmedZ I need it because of dependency conflict

**Comment by rrousselGit** [comment]
What dependency conflict?

**Comment by amorenew** [comment]
@rrousselGit First step I upgrade to Flutter 1.22
then `flutter pub run build_runner build --delete-conflicting-outputs`
Result:
```
[WARNING] freezed:freezed on test/unit/infrastructure/core/http/api_interceptor_test.dart:
Your current `analyzer` version may not fully support your current SDK version.

Please try upgrading to the latest `analyzer` by running `flutter packages upgrade`.

Analyzer language version: 2.9.0
SDK language version: 2.10.0

If you are getting this message and have the latest analyzer please file
an issue at https://github.com/dart-lang/sdk/issues/new with the title
"No published analyzer available for language version 2.10.0".
Please search the issue tracker first and thumbs up and/or subscribe to
existing issues if present to avoid duplicates.

[INFO] Generating SDK summary...
[SEVERE] freezed:freezed on test/unit/infrastructure/core/http/api_interceptor_test.dart:

Bad state: Unexpected diagnostics:
/Users/apple/Desktop/sdks/flutter/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:152:17 - This requires the 'non-nullable' language feature to be enabled.
/Users/apple/Desktop/sdks/flutter/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:119:41 - This requires the 'non-nullable' language feature to be enabled.
```

Then I run `flutter pub upgrade` and added in analysis_options.yaml
```
analyzer:
  enable-experiment:
    - non-nullable
```
It works fine

My issue is that I am using Codemagic CI so how could I skip `flutter pub upgrade`?

**Comment by amorenew** [comment]
@salehahmedZ @rrousselGit I solved my issue
```
environment:
  sdk: ">=2.10.0 <3.0.0"
dependency_overrides:
  analyzer: ^0.39.17

```
More info https://github.com/flutter/flutter/issues/62240

**Comment by Joezzy** [comment]
> @rrousselGit First step I upgrade to Flutter 1.22
> Then I run `flutter pub run build_runner build --delete-conflicting-outputs`
> Result:
> 
> ```
> [WARNING] freezed:freezed on test/unit/infrastructure/core/http/api_interceptor_test.dart:
> Your current `analyzer` version may not fully support your current SDK version.
> 
> Please try upgrading to the latest `analyzer` by running `flutter packages upgrade`.
> 
> Analyzer language version: 2.9.0
> SDK language version: 2.10.0
> 
> If you are getting this message and have the latest analyzer please file
> an issue at https://github.com/dart-lang/sdk/issues/new with the title
> "No published analyzer available for language version 2.10.0".
> Please search the issue tracker first and thumbs up and/or subscribe to
> existing issues if present to avoid duplicates.
> 
> [INFO] Generating SDK summary...
> [SEVERE] freezed:freezed on test/unit/infrastructure/core/http/api_interceptor_test.dart:
> 
> Bad state: Unexpected diagnostics:
> /Users/apple/Desktop/sdks/flutter/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:152:17 - This requires the 'non-nullable' language feature to be enabled.
> /Users/apple/Desktop/sdks/flutter/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:119:41 - This requires the 'non-nullable' language feature to be enabled.
> ```
> 
> Then I run `flutter pub upgrade` and added in analysis_options.yaml
> 
> ```
> analyzer:
>   enable-experiment:
>     - non-nullable
> ```
> 
> and in pubspec.yaml
> 
> ```
> environment:
>   sdk: ">=2.10.0 <3.0.0"
> ```
> 
> It works fine
> 
> My issue is that I am using Codemagic CI so how could I skip `flutter pub upgrade`?

this worked for me
